compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1065,7 +1065,7 @@ object desugar {
             AppliedTypeTree(ref(seqType), t),
             New(ref(defn.RepeatedAnnotType), Nil :: Nil))
         } else {
-          assert(ctx.mode.isExpr || ctx.reporter.hasErrors, ctx.mode)
+          assert(ctx.mode.isExpr || ctx.reporter.hasErrors || ctx.mode.is(Mode.Interactive), ctx.mode)
```

Comment:
Any idea why we manage to hit this assertion in the IDE?

Reply:
No idea, I am afraid.